### PR TITLE
[multigrid-nonsymmetric-smoother] Allow nonsymmetric smoothers in Multigrid object.

### PR DESCRIPTION
**Pull request by barker29**
A small modification to the `Multigrid::SmoothingStep` method and a few related places, so that the `Multigrid` object can now do a (symmetric) V-cycle by doing `smoother->Mult` on the way down and `smoother->MultTranspose` on the way up.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1582](https://github.com/mfem/mfem/pull/1582) | @barker29 | @v-dobrev | @drzisga + @psocratis | 07/01/20 | 07/15/20 | 07/21/20 | |

### CODE REVIEW

File: linalg/solvers.hpp [hunk 1]
```diff
@@ -181,6 +182,8 @@ class OperatorChebyshevSmoother : public Solver
 
    void Mult(const Vector&x, Vector &y) const;
 
+   void MultTranspose(const Vector &x, Vector &y) const { Mult(x, y); }
```
Comment (drzisga) [inline linalg/solvers.hpp:185]: I am not sure if the Chebyshev smoother is symmetric as it is implemented now. A Chebyshev polynomial smoother requires, in general, a matrix which is scaled such that the entries on the diagonal are one. Here, we work with `D^(-1) * A` which is non-symmetric, but in theory we would rather want to use `D^(-1/2) * A * D^(-1/2)`.

In practice, this doesn't seem to be an issue. Do you maybe know more about this?

Comment (barker29) [reply linalg/solvers.hpp:185]: Looking at the code in `OperatorChebyshevSmoother::Mult`, I get
```
y_0 = c_0 D^{-1} b
y_1 = (c_0 D^{-1} + c_1 D^{-1} A D^{-1}) b
y_2 = (c_0 D^{-1} + c_1 D^{-1} A D^{-1} + c_2 D^{-1} A D^{-1} A D^{-1}) b
```
These all look symmetric to me, but I might be missing something?

Comment (drzisga) [reply linalg/solvers.hpp:185]: You are correct. I was thinking too complicated and made a mistake. Your changes look good to me.

**Comment by v-dobrev** [comment]
Merged in `next` for testing...

**Pull request merged**